src/main/java/com/emc/ecs/nfsclient/nfs/io/NfsFileBase.java

```diff
@@ -278,6 +290,82 @@ public boolean exists() throws IOException {
         return exists;
     }
 
+    /* (non-Javadoc)
+     * @see com.emc.ecs.nfsclient.nfs.io.NfsFile#followLinks()
+     */
+    public F followLinks() throws IOException {
+        if (_backingFile == null) {
+            F backingFile = (F) this;
+            Set<String> pathsTried = new HashSet<String>();
+            int linksTraversed = 0;
+            NfsGetAttributes attributes = safeGetAttributes(backingFile);
+            while (NfsType.NFS_LNK == attributes.getType()) {
+                pathsTried.add(backingFile.getPath());
+                backingFile = followLink(backingFile.readlink().getData());
+                if (pathsTried.contains(backingFile.getPath())) {
+                    throw new IOException("Links form a loop: " + backingFile.getPath());
+                }
+                ++linksTraversed;
+                if (linksTraversed > 100) {
+                    throw new IllegalArgumentException("Too many links to follow (> 100).");
+                }
+                attributes = safeGetAttributes(backingFile);
+            }
+            _backingFile = backingFile;
+        }
+
+        return _backingFile;
+    }
+
+    /**
+     * @param backingFile
```

Comment:
Good point. This is a tricky bit, I just realized that this loop prevention algorithm is probably not quite right. rethinking...